### PR TITLE
Implement Data Filter Ignored Queries

### PR DESCRIPTION
**Pull request by ndegwamartin**
- [x] Skip Data Sync Filter
- [x] Generic Configuration file
- [x] Configure Skipping for composition children Questionnaires, Structure Maps and List Resources
- [x] Unit Tests
- [x] Documentation

### CODE REVIEW

File: plugins/src/main/java/com/google/fhir/proxy/plugin/OpenSRPSyncAccessDecision.java [hunk 1]
```diff
@@ -193,4 +206,108 @@ private boolean isResourceTypeRequest(String requestPath) {
...
+            return false;
+          }
+
+          if (expectedParam.getValue() instanceof List) {
+
+            if (CollectionUtils.isEqualCollection(
+                (List) expectedParam.getValue(), Arrays.asList(actualQueryValue[0].split(",")))) {
+              return true;
+            } else {
+              return false;
```
Comment (dubdabasoduba) [inline plugins/src/main/java/com/google/fhir/proxy/plugin/OpenSRPSyncAccessDecision.java:279]: I think we can simplify this check.  `return CollectionUtils.isEqualCollection(
                (List) expectedParam.getValue(), Arrays.asList(actualQueryValue[0].split(",")))`

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]
Just a minor cosmetic change

File: plugins/src/main/java/com/google/fhir/proxy/plugin/OpenSRPSyncAccessDecision.java (hunk 1 above)
Comment (ndegwamartin) [reply plugins/src/main/java/com/google/fhir/proxy/plugin/OpenSRPSyncAccessDecision.java:279]: I'm open to ideas. Basically we have two lists and we want to ensure that the values they contain are exactly the same in which ever order.

Comment (dubdabasoduba) [reply plugins/src/main/java/com/google/fhir/proxy/plugin/OpenSRPSyncAccessDecision.java]: Made the update. Upgraded the tag too

**Review by dubdabasoduba — APPROVED** [review]

File: plugins/src/main/java/com/google/fhir/proxy/plugin/OpenSRPSyncAccessDecision.java (hunk 1 above)
Comment (ndegwamartin) [reply plugins/src/main/java/com/google/fhir/proxy/plugin/OpenSRPSyncAccessDecision.java]: Looks good, more concise 👍🏾

**Pull request merged**